frontend/projects/upgrade/src/app/features/dashboard/feature-flags/modals/upsert-feature-flag-modal/upsert-feature-flag-modal.component.ts

```diff
@@ -106,6 +106,23 @@ export class UpsertFeatureFlagModalComponent {
     this.listenForPrimaryButtonDisabled();
     this.listenForDuplicateKey();
     this.listenOnContext();
+
+    if (this.isDisabled()) {
+      this.disableRestrictedFields();
+    }
+  }
+
+  isDisabled() {
+    return (
+      this.config.params.action === UPSERT_FEATURE_FLAG_ACTION.EDIT &&
+      this.config.params.sourceFlag?.status === FEATURE_FLAG_STATUS.ENABLED
+    );
+  }
+
+  disableRestrictedFields(): void {
```

Comment:
Do you need this function? You are already disabling the form with this code `[disabled]="isDisabled()"`